risc0/zkvm/platform/src/syscall.rs

```diff
@@ -263,3 +268,25 @@ pub unsafe fn sys_sha_buffer(
     #[cfg(not(target_os = "zkvm"))]
     unimplemented!()
 }
+
+#[inline(always)]
+pub unsafe fn sys_rand(recv_bytes: usize) -> &'static [u8] {
```

Comment:
I think we should pass a pointer to be filled in here at this level (the syscall level). This would allow the use of custom allocators and doesn't assume any allocations in `risc0-zkvm-platform`.

Reply:
Also, using a pointer sets us up for potential use from other languages.